### PR TITLE
- updated appauth library to a separate module.

### PR DESCRIPTION
**Pull request by PINAKIN-KANSARA-EY**
In order to improve the overall authentication handling in the application. this is the first step in the direction.
decoupled app auth library from the repository to an individual module.

### CODE REVIEW

File: appauth/build.gradle [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+}
+
+dependencies {
+
+    implementation "androidx.appcompat:appcompat:$versions.appcompat"
+    api "androidx.browser:browser:1.3.0"
```
Comment (bsavoini-fw) [inline appauth/build.gradle:32]: Please use the versions declared in the versions.gradle file

File: repository/src/main/java/ca/bc/gov/repository/bcsc/BcscAuthRepo.kt [hunk 2]
```diff
@@ -36,11 +28,11 @@ class BcscAuthRepo(
     private val patientLocalDataSource: PatientLocalDataSource,
 ) {
 
-    private lateinit var authState: AuthState
-    private lateinit var authService: AuthorizationService
-    private lateinit var authServiceConfiguration: AuthorizationServiceConfiguration
+    private lateinit var authState: net.openid.appauth.AuthState
```
Comment (bsavoini-fw) [inline repository/src/main/java/ca/bc/gov/repository/bcsc/BcscAuthRepo.kt:31]: Is it possible to declare the imports in the top for  AuthState, AuthorizationService, AuthorizationServiceConfiguration, etc?

File: repository/src/main/java/ca/bc/gov/repository/bcsc/BcscAuthExtension.kt [hunk 3]
```diff
@@ -19,10 +13,10 @@ import kotlin.coroutines.resumeWithException
 * Created by amit_metri on 31,January,2022
 */
 
-suspend fun awaitFetchFromIssuer(uri: Uri): AuthorizationServiceConfiguration =
+suspend fun awaitFetchFromIssuer(uri: Uri): net.openid.appauth.AuthorizationServiceConfiguration =
     suspendCancellableCoroutine { continuation ->
 
-        AuthorizationServiceConfiguration.fetchFromIssuer(
+        net.openid.appauth.AuthorizationServiceConfiguration.fetchFromIssuer(
```
Comment (bsavoini-fw) [inline repository/src/main/java/ca/bc/gov/repository/bcsc/BcscAuthExtension.kt:19]: Is it possible to declare the imports in the top for  AuthorizationServiceConfiguration, etc?

**Review by bsavoini-fw — CHANGES_REQUESTED** [review]

File: appauth/build.gradle (hunk 1 above)
Comment (PINAKIN-KANSARA-EY) [reply appauth/build.gradle]: done

File: repository/src/main/java/ca/bc/gov/repository/bcsc/BcscAuthRepo.kt (hunk 2 above)
Comment (PINAKIN-KANSARA-EY) [reply repository/src/main/java/ca/bc/gov/repository/bcsc/BcscAuthRepo.kt]: updated, auto import cause this issue

File: repository/src/main/java/ca/bc/gov/repository/bcsc/BcscAuthExtension.kt (hunk 3 above)
Comment (PINAKIN-KANSARA-EY) [reply repository/src/main/java/ca/bc/gov/repository/bcsc/BcscAuthExtension.kt]: updated

**Review by bsavoini-fw — APPROVED** [review]

**Pull request merged**